### PR TITLE
Separate Makefile targets for local vs containerized

### PR DESCRIPTION
**Pull request by SudoBrendan**
### Which issue this PR addresses:

Fixes feedback/concerns raised by the wider ARO team.

Loki is committed to enhancing our build and release process by moving local testing and validation closer to a production-like state. While we believe this approach will effectively address current issues, we respect individual preferences and will not mandate its use. Loki will continue to elevate our CI and containerized builds, ensuring top-tier local execution. Our CI/CD roadmap will continue to focus on containerized builds and validation for both PR validation and production builds, because our current architecture runs containers in production as defined in these build processes.

TL;DR - you now need to opt into the new, containerized build process locally. There is a clear distinction and naming consistency between local binary and containerized build/execution. Use whatever you feel is best locally.

### What this PR does / why we need it:

Provides flexibility in local development - you can now opt into containerized build/run commands. Old commands are 100% reverted.

### Test plan for issue:

Only affects local.

### Is there any documentation that needs to be updated for this PR?

Yes, included in the PR.

### How do you know this will function as expected in production? 

N/A

### CODE REVIEW

**Review by tsatam — DISMISSED** [review]

**Review by shubhadapaithankar — DISMISSED** [review]

**Review by kevinobriendotca — APPROVED** [review]

**Review by jhoreman — APPROVED** [review]
Approved, thanks for this.

**Review by kevinobriendotca — APPROVED** [review]

**Review by tsatam — APPROVED** [review]
